core/network/src/on_demand.rs

```diff
@@ -390,12 +408,33 @@ impl<B, E> OnDemandCore<B, E> where
 			None => return,
 		};
 
+		let last_peer = self.idle_peers.back().cloned();
 		while !self.pending_requests.is_empty() {
 			let peer = match self.idle_peers.pop_front() {
 				Some(peer) => peer,
 				None => return,
 			};
 
+			// check if request can (optimistically) be processed by the peer
+			if {
```

Comment:
this would be more readable if the condition was assigned to a temporary rather than the weird `if { } { }`